drivers/hmis/app/models/hmis/user_group.rb

```diff
@@ -0,0 +1,60 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::UserGroup < ApplicationRecord
+  acts_as_paranoid
+  has_paper_trail
+  include UserPermissionCache
+
+  has_many :access_controls, inverse_of: :user_group
+  has_many :user_group_members, class_name: '::Hmis::UserGroupMember', inverse_of: :user_group
+  has_many :users, through: :user_group_members
+
+  after_save :invalidate_user_permission_cache
+
+  scope :not_system, -> do
+    where(system: false)
+  end
+
+  # hide previous declaration of :system (from Kernel), we'll use this one
+  replace_scope :system, -> do
+    where(system: true)
+  end
+
+  scope :with_user_id, ->(user_id) do
+    joins(:user_group_members).
+      merge(Hmis::UserGroupMember.where(user_id: user_id))
+  end
+
+  def self.system_user
```

Comment:
nit: can we call this `system_user_group` since its returning a group, not a user

Reply:
Yes, and I don't think we have any usages of that yet, so I'm not quite sure why we have it, but I'll rename and leave for now.  Pretty sure there will be reasons.